### DISCUSSION TITLE
使用dynv6的二级域名时callback报错

### DISCUSSION THREAD

**Discussion by SnowRain1326** [jeessy2/ddns-go#1150] [Q&A] [answered]
根据Callback配置参考写出的https://dynv6.com/api/update?zone=#{domain}&token=YOUR_TOKEN&ipv4=#{ip}&ipv6=#{ip}
在保存后提示token for authorization missing，返回状态码401，其中的domain和token都配置正确，但是无法更新

**Comment by jeessy2** [comment]
token对不对

**Reply by SnowRain1326** [reply]
token我确定是对的 因为一直报错反复复制了好几遍，而且仔细对照过

**Reply by jeessy2** [reply]
&ipv6移除呢

**Reply by SnowRain1326** [reply]
> &ipv6移除呢

移除了也是报同样的错

**Reply by jeessy2** [reply]
那你参考一下对方官方的文档重新写一个

**Reply by SnowRain1326** [reply]
用官方文档改写之后提示callback调用失败，异常信息：返回内容400 bad request，返回状态码400

**Reply by WaterLemons2k** [reply]
再用 `curl -v` 测试一遍 dynv6 的 URL；看最后请求的路径，可能是因为有内容被错误转义了。

**Reply by SnowRain1326** [reply]
> 再用 `curl -v` 测试一遍 dynv6 的 URL；看最后请求的路径，可能是因为有内容被错误转义了。

修改了一下之后现在报的错是 callback调用失败，invalid ipv6 address 返回状态码422 我尝试过换别的方法获取IP也不行 选择只获取ipv4之后他也报invalid ipv4 address

**Comment by SnowRain1326** [comment, marked as answer]
解决了，自己疯狂试错找到了正确的格式
https://dynv6.com/api/update?zone=你的域名&token=对应密钥&ipv4&ipv6prefix
![Uploading 屏幕截图 2024-06-12 105315.png…]()
